### ISSUE TITLE
Exoplayer FFMPEG Extention throws error 

### ISSUE THREAD

**Issue by ThreeMinutesFix** [google/ExoPlayer#10774]
Good days team and dev,
i am developing android app and implementing 5.1 DD support so using FFMPEG extention i got it worked with exo 2.18.1 but when  i try to do that on my app which use 2.15 version  it thorws error
`My exo version   implementation 'com.google.android.exoplayer:exoplayer:2.15.1'`
Current implementation
```
ExoTrackSelection.Factory videoTrackSelectionFactory = new AdaptiveTrackSelection.Factory();
        trackSelector = new DefaultTrackSelector(Player.this, videoTrackSelectionFactory);
        DefaultRenderersFactory renderersFactory = new DefaultRenderersFactory(this);

renderersFactory.setExtensionRendererMode(DefaultRenderersFactory.EXTENSION_RENDERER_MODE_PREFER);
        simpleExoPlayer = new SimpleExoPlayer.Builder(thisrenderersFactory).setTrackSelector(trackSelector).setSeekForwardIncrementMs(10000).setSeekBackIncrementMs(10000).build();
        playerView.setPlayer(simpleExoPlayer);

        playerView.setKeepScreenOn(true);

```
If extention work only after v18  please tell me the change version of code because if i upgrade exoplayer tells to change all the depreciation values 
including

```
DefaultHttpDataSourceFactory httpDataSourceFactory = new DefaultHttpDataSourceFactory(userAgent, null /* listener */, DefaultHttpDataSource.DEFAULT_CONNECT_TIMEOUT_MILLIS, DefaultHttpDataSource.DEFAULT_READ_TIMEOUT_MILLIS, true /* allowCrossProtocolRedirects */)

        if (!defaultRequestProperties.isEmpty()) {
            httpDataSourceFactory.setDefaultRequestProperties(defaultRequestProperties);
        }

        DefaultDataSourceFactory dataSourceFactory = new DefaultDataSourceFactory(context, null /* listener */, httpDataSourceFactory);
```

I request anyone to help me
What i tried:
stackoverflow (outdated)
Google demo app (not relevent totally different of my implementation)
finally my only hope is this platform

**Comment by ThreeMinutesFix** [comment]
Forgot to mention the error

`E/AndroidRuntime: FATAL EXCEPTION: ExoPlayer:Playback
    Process: com.gospel.kids, PID: 25765
    java.lang.VerifyError: Superclass com.google.android.exoplayer2.util.LibraryLoader of com.google.android.exoplayer2.ext.ffmpeg.FfmpegLibrary$1 is declared final (declaration of 'com.google.android.exoplayer2.ext.ffmpeg.FfmpegLibrary$1' appears in /data/app/com.gospel.kids-CBVa37n2L5fnFeHda_kB5Q==/base.apk!classes3.dex)
        at com.google.android.exoplayer2.ext.ffmpeg.FfmpegLibrary.isAvailable(FfmpegLibrary.java:61)
        at com.google.android.exoplayer2.ext.ffmpeg.FfmpegAudioRenderer.supportsFormatInternal(FfmpegAudioRenderer.java:93)
        at com.google.android.exoplayer2.audio.DecoderAudioRenderer.supportsFormat(DecoderAudioRenderer.java:218)
        at com.google.android.exoplayer2.trackselection.MappingTrackSelector.findRenderer(MappingTrackSelector.java:485)
        at com.google.android.exoplayer2.trackselection.MappingTrackSelector.selectTracks(MappingTrackSelector.java:355)
        at com.google.android.exoplayer2.MediaPeriodHolder.selectTracks(MediaPeriodHolder.java:238)
        at com.google.android.exoplayer2.MediaPeriodHolder.handlePrepared(MediaPeriodHolder.java:186)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handlePeriodPrepared(ExoPlayerImplInternal.java:2202)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:501)
        at android.os.Handler.dispatchMessage(Handler.java:103)
        at android.os.Looper.loop(Looper.java:225)
        at android.os.HandlerThread.run(HandlerThread.java:67)`

**Comment by ThreeMinutesFix** [comment]
actually i solved the problem now i can play 5.1 and 7.1 audio too. reason is because of older version player thrown error
